packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -159,18 +162,33 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
     // Signer needs to be initialised here before defaultValidationModule is set
     if (biconomySmartAccountConfig.signer) {
       const signer = biconomySmartAccountConfig.signer;
-      // AA's WalletClientSigner has a signerType, viems' walletClient does not:
-      const isViemWalletClient = !(signer as WalletClientSigner)?.signerType;
-      if (isViemWalletClient) {
-        const walletClient = signer as WalletClient;
-        if (!walletClient.account) {
-          throw new Error("Cannot consume a viem wallet without an account");
-        }
-        if (!walletClient.chain) {
-          throw new Error("Cannot consume a viem wallet without a chainId");
+
+      const isAnAlchemySmartWallet = UNIQUE_PROPERTIES_PER_SIGNER.AlchemySmartAccountSigner in signer;
```

Comment:
Could use `instanceof SmartAccountSigner` here (and`instanceof Signer`, `instanceof WalletClient`) but not sure how consistent this check is on the instance across different package versions 🤔.